src/neptune/api/searching_entries.py

```diff
@@ -148,34 +181,45 @@ def iter_over_pages(
     max_offset: int = MAX_SERVER_OFFSET,
     sort_by_column_type: Optional[str] = None,
     ascending: bool = False,
+    progress_bar: Optional[Union[bool, Type[ProgressBarCallback]]] = None,
     **kwargs: Any,
 ) -> Generator[Any, None, None]:
     searching_after = None
     last_page = None
 
-    while True:
-        if last_page:
-            page_attribute = find_attribute(entry=last_page[-1], path=sort_by)
+    total = fetch_data(
+        limit=0,
+        offset=0,
+        **kwargs,
+    ).get("matchingItemCount", 0)
 
-            if not page_attribute:
-                raise ValueError(f"Cannot find attribute {sort_by} in last page")
+    progress_bar = progress_bar if step_size >= total else None
+    with which_progress_bar(progress_bar)(description="Fetching table...") as bar:
+        while True:
+            if last_page:
+                page_attribute = find_attribute(entry=last_page[-1], path=sort_by)
 
-            searching_after = page_attribute.properties["value"]
+                if not page_attribute:
+                    raise ValueError(f"Cannot find attribute {sort_by} in last page")
 
-        for offset in range(0, max_offset, step_size):
-            page = get_single_page(
-                limit=min(step_size, max_offset - offset),
-                offset=offset,
-                sort_by=sort_by,
-                sort_by_column_type=sort_by_column_type,
-                searching_after=searching_after,
-                ascending=ascending,
-                **kwargs,
-            )
+                searching_after = page_attribute.properties["value"]
+
+            for offset in range(0, max_offset, step_size):
+                page = get_single_page(
+                    limit=min(step_size, max_offset - offset),
+                    offset=offset,
+                    sort_by=sort_by,
+                    sort_by_column_type=sort_by_column_type,
+                    searching_after=searching_after,
+                    ascending=ascending,
+                    **kwargs,
+                )
+
+                if not page:
+                    return
 
-            if not page:
-                return
+                bar.update(by=step_size, total=total)
```

Comment:
We have to support 10k+ entries but to my knowledge `matchingItemCount` will be equal to 10k at most. You have to take into account this value each time a new 10k page is started.